vyper/codegen/expr.py

```diff
@@ -279,6 +279,8 @@ def parse_Attribute(self):
                 return IRnode.from_list(["callvalue"], typ=BaseType("uint256"))
             elif key == "msg.gas":
                 return IRnode.from_list(["gas"], typ="uint256")
+            elif key == "block.prevrandao":
+                return IRnode.from_list(["difficulty"], typ="uint256")
```

Comment:
Put a note here to change post-merge

Reply:
https://github.com/vyperlang/vyper/pull/3085/commits/30149a8ff3f638b58b457d4a10a4153cbdd529e4

Reply:
i think actually what we want is for `block.prevrandao` to compile to the `PREVRANDAO` mnemonic. they will still assemble to the same bytecode but which one was used will carry through all the way to assembly, which could be useful for debugging.

alternatively, we can add a `version_check()` rule which forces both to compile to prevrandao or difficulty, depending on the fork.

Reply:
fixed here https://github.com/vyperlang/vyper/pull/3085/commits/fccc9e0bad3d6f9ec6c9172d904a1ba3e93f1746. If you agree, please resolve the conversation.